### PR TITLE
Gutenberg: Fix Publicize block again

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

* Add Publicize block to the Jetpack preset again.
* Use `@wordpress/edit-post`.
* Fix usage of lodash
* Use `apiFetch` instead of `wp.apiRequest` and `jQuery.post()`
* Sort imports.
* Ditch any usage of the global `wp` var.
* Ditch any usage of the global `$/jQuery` var.
* Small formatting fixes.

#### Testing instructions

* Checkout this branch.
* Spin up local Jetpack with the `try/publicize-gutenberg-block-externally-built` branch.
* Make sure Jetpack is active and connected.
* Run `npm run sdk gutenberg client/gutenberg/extensions/presets/jetpack -- --output-dir=DIR_TO_JETPACK/jetpack/_inc/blocks` where `DIR_TO_JETPACK` is your Jetpack directory.
* Start writing a new post and attempt to publish.
* Verify Publicize appears in the pre-publish panel.
* Verify Publicize appears in the plugin sidebar - there should be a Jetpack icon in the toolbar - click it to reveal the Publicize UI.

### CODE REVIEW

File: client/gutenberg/extensions/publicize/form.jsx [hunk 1]
```diff
@@ -11,8 +11,8 @@
 /**
  * External dependencies
  */
+import isNil from 'lodash/isNil';
```
Comment (enejb) [inline client/gutenberg/extensions/publicize/form.jsx]: Not really part of this PR but can we just remove the dependence on lodash here 
since the isNil function is just 
```
function isNil(value) {
      return value == null;
}``` 
https://github.com/lodash/lodash/blob/4.17.10/lodash.js#L11972

**Comment by ockham** [comment]
Update, I got #10372 to work. It should now be possible to test this PR with Jetpack's `try/publicize-gutenberg-block-externally-built-rebased` branch.

File: client/gutenberg/extensions/publicize/form.jsx (hunk 1 above)
Comment (tyxla) [reply client/gutenberg/extensions/publicize/form.jsx]: 👍Good idea - I'll follow-up in another PR.

File: package.json [hunk 2]
```diff
@@ -45,6 +45,7 @@
     "@wordpress/data": "2.1.1",
     "@wordpress/date": "2.0.2",
     "@wordpress/deprecated": "2.0.2",
+    "@wordpress/edit-post": "1.0.4",
```
Comment (ockham) [inline package.json]: Hmm, if we're not really needing this for Calypso but only for our `jetpack-blocks` npm, it should arguably be in https://github.com/Automattic/wp-calypso/blob/87037c078075b7e838146c060069f565161a95d4/client/gutenberg/extensions/presets/jetpack/package.json...

Comment (ockham) [reply package.json]: I'm not going to block this PR on this since right now, we don't have any dependencies in `jetpack-blocks`' `package.json` anyway. It's something we should do tho (in a separate PR).

Comment (ockham) [reply package.json]: /cc @sirreal @simison

Comment (vindl) [reply package.json]: We'll probably be needing this in Calypso soon, so I think it's fine if it remains here.

**Comment by vindl** [comment]
Hi @tyxla, this import is breaking the editor locally for me:

`import 'gutenberg/extensions/publicize/editor';`

it's also broken now in `wpcalypso` (at least for all of my test sites).

**Comment by tyxla** [comment]
:/

Yeah, we're still working on it, and it'll probably need some sort of feature flag, as it'll only work on Jetpack sites for now. Disabling it for the time being: #28015
